### PR TITLE
Rqt py common service

### PR DESCRIPTION
**Pull request by mlautman**
some helper functions for rqt_msg and rqt_srv.

### CODE REVIEW

File: rqt_py_common/src/rqt_py_common/message_helpers.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+from ament_index_python import get_resource
+from ament_index_python import get_resources
+from ament_index_python import has_resource
+
+
+MSG_MODE = 'msg'
+SRV_MODE = 'srv'
+ACTION_MODE = 'action'
+
+# Taken from https://github.com/ros2/ros2cli/blob/bouncy/ros2msg/ros2msg/api/__init__.py
```
Comment (brawner) [inline rqt_py_common/src/rqt_py_common/message_helpers.py:42]: If taking code from a different file, do you preserve the license of the package you are installing it into or the license fo the file you took it from?

**Review by brawner — COMMENTED** [review]
As most of this code comes from elsewhere, it looks good to me. Can you link to a passing CI test. This doesn't pass tests on my machine

**Comment by dirk-thomas** [comment]
The logic around `.msg` and `.srv` files is going to break with the upcoming IDL changes. Maybe delaying this for now would be good to not waste effort.

File: rqt_py_common/src/rqt_py_common/message_helpers.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+def get_all_service_types():
```
Comment (brawner) [inline rqt_py_common/src/rqt_py_common/message_helpers.py:43]: Two lines of space above function definition

File: rqt_py_common/src/rqt_py_common/topic_helpers.py [hunk 3]
```diff
@@ -115,6 +115,52 @@ def get_message_class(message_type):
     return class_val
 
 
+_service_class_cache = {}
+def get_service_class(service_type):
```
Comment (brawner) [inline rqt_py_common/src/rqt_py_common/topic_helpers.py:119]: Two lines of space above function definition

File: rqt_py_common/src/rqt_py_common/topic_helpers.py [hunk 4]
```diff
@@ -115,6 +115,52 @@ def get_message_class(message_type):
...
+    logger = logging.get_logger("get_service_class")
+    if service_type in _service_class_cache:
+        return _service_class_cache[service_type]
+
+    service_info = service_type.split("/")
```
Comment (brawner) [inline rqt_py_common/src/rqt_py_common/topic_helpers.py:124]: Single quotes
```suggestion
    service_info = service_type.split('/')
```

**Comment by mlautman** [comment]
Closing in favor of #162

**Pull request closed without merging**